ittapi-rs/integration-tests/src/main.rs

```diff
@@ -0,0 +1,17 @@
+use ittapi::{Domain, Task};
+use std::{thread::sleep, time::Duration};
+
+fn main() {
+    let task_duration = Duration::from_millis(50);
+    let domain = Domain::new("test");
+
+    let task1 = Task::begin(&domain, "task#1");
+    println!("Running task#1...");
+    sleep(task_duration);
+    task1.end();
```

Comment:
> if the ITT dynamic library (actual collector) is absent, the domain and the task1 will be invalid. Correct?

Yes, correct. 

> Will this code work correct in case of invalid domain and task1? Do we need additional checks to filter our the invalid situation or something?

Right now the check that is done is "if the ITT function pointer is `None`, then don't call any ITT function." From your previous explanations, I thought this is the most general way to make this library never fail. But this assumption would be invalid if, e.g., the collector later wires up some of those function pointers but the user has already created an invalid domain--then, the next function call would go through with an invalid domain. Is this ever possible?

But I think this would still work, since the C side will check if the parameters are null pointers. Am I right?